dask_jobqueue/oar.py

```diff
@@ -82,33 +81,41 @@ def __init__(
 
         # Memory
         if self.worker_memory is not None:
-            if mem_core_property_name is None:
-                mem_core_property_name = dask.config.get(
-                    "jobqueue.%s.mem-core-property-name" % self.config_name
+            if memory_per_core_property_name is None:
+                memory_per_core_property_name = dask.config.get(
+                    "jobqueue.%s.memory-per-core-property-name" % self.config_name
                 )
-            if mem_core_property_name is None:
+            if memory_per_core_property_name is None:
                 warn = (
                     "OAR Job memory reserved resources will not be set according to Dask Worker memory limit, "
-                    "which can cause crashes."
+                    "which can cause crashes. See `memory_per_core_property_name` documentation for more details"
                 )
                 warnings.warn(warn, category=UserWarning)
             else:
-                # OAR expects MiB as memory unit
-                oar_memory = int(
-                    parse_bytes(self.worker_memory / self.worker_cores) / 2**20
-                )
-                # OAR needs to have the properties on a single line, with SQL syntax
-                # If there are several "#OAR -p" lines, only the last one will be taken into account by OAR
-                last_job_property = return_last_job_property(self.job_extra_directives)
-                if last_job_property is not None:
-                    header_lines.append(
-                        "#OAR -p '%s AND %s>=%s'"
-                        % (last_job_property, mem_core_property_name, oar_memory)
+                if memory_per_core_property_name != "not_applicable":
+                    # OAR expects MiB as memory unit
+                    oar_memory = math.ceil(
```

Comment:
I use `math.ceil` here to be sure that the rounding is done towards bigger values, so that the allocated nodes has a bit more memory that the Dask worker memory limit